src/libpgmoneta/configuration.c

```diff
@@ -3021,3 +3041,45 @@ is_empty_string(char* s)
 
    return true;
 }
+
+static int
+remove_leading_whitespace(const char* s, char** trimmed_line)
+{
```

Comment:
Please, use`pgmoneta_append_char()` and assign `trimmed_line` if successful, and `return 0`.

Otherwise `free()` and `return 1`.

You know that the line will either be a comment, or a `key = value` line with maybe a comment at the end